### PR TITLE
create erc20 contract

### PR DESCRIPTION
**Pull request by rccsousa**
## Why:
Players that are willing to play Wordle3 will have to use `WDL` ERC20 tokens.
## How:
By implementing a standard ERC20 token that will be used to play Wordle3.

### CODE REVIEW

File: script/ERC20.s.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import {WDLToken} from "../src/ERC20.sol";
+
+contract ERC20Script is Script {
+    function run() external {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        vm.startBroadcast(deployerPrivateKey);
+
+        // Deploy the token
+        WDLToken token = new WDLToken(1000);
+
```
Comment (rccsousa) [inline script/ERC20.s.sol:14]: Wordle deployment can then go here when once token is implemented on the Wordle contract, correct?
something along the lines of `Wordle wordle = new Wordle(word, tokenAddress)`  

also word is declared on the `.env` file to keep it hidden?

Comment (rccsousa) [reply script/ERC20.s.sol:14]: UPDATE: after further researching this comment does not make sense.
Using the private key makes sense for deploying the token and storing the address for the Wordle contract deployment but having it here does not.

File: src/ERC20.sol [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract WDLToken is ERC20 {
+    constructor(uint256 initialSupply) ERC20("Wordle Token", "WDL") {
+        _mint(msg.sender, initialSupply * 10 ** 18);
+    }
```
Comment (rccsousa) [inline src/ERC20.sol:9]: Would it be a good idea to:
1. hard code supply;
2. switch to `ERC20Capped`;

If I understood correctly we can also declare the address that will be used on deployment to facilitate calling said address when deploying the game, correct?

File: script/ERC20.s.sol (hunk 1 above)
Comment (DavideSilva) [reply script/ERC20.s.sol:14]: The deploy script should probably be renamed to `Deploy.s.sol` and take care of deploying both contracts. It's ok to have individual scripts that just deploy one contract but you probably want to have a script that deploys all that is needed for the project

File: src/ERC20.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (rccsousa) [reply src/ERC20.sol:9]: ok, so leaving as is for the time being

Comment (DavideSilva) [reply src/ERC20.sol:9]: Yes, ssers of the game should be able to call mint directly on the ERC20 contract to get some tokens in order to play

**Review by DavideSilva — APPROVED** [review]
Check the comments and :shipit:

File: script/ERC20.s.sol (hunk 1 above)
Comment (rccsousa) [reply script/ERC20.s.sol]: renamed on https://github.com/subvisual/wordle3/pull/8/commits/a3e973affd650e1e0baf171fddb8028fbf4b3755 ✅

**Pull request merged**